packages/cdk-api-gateway-domain/lib/index.ts

```diff
@@ -0,0 +1,166 @@
+import * as cdk from '@aws-cdk/core';
+import * as apigw from '@aws-cdk/aws-apigateway';
+import * as acm from '@aws-cdk/aws-certificatemanager';
+import * as iam from '@aws-cdk/aws-iam';
+import * as route53 from '@aws-cdk/aws-route53';
+import * as targets from '@aws-cdk/aws-route53-targets';
+import * as customresource from '@aws-cdk/custom-resources'
+
+export interface CdkApiGatewayDomainProps {
+  companyDomainName: string;
+  companyHostedZoneId: string;
+  project: string;
+  stage: string;
+  certificateArn: string;
+  restApiId: string;
+  basePath: string;
+}
+
+export class CdkApiGatewayDomain extends cdk.Construct {
+
+  constructor(scope: cdk.Construct, id: string, props: CdkApiGatewayDomainProps) {
+    super(scope, id);
+
+    const api = apigw.RestApi.fromRestApiId(this, "Rest API", `${props.restApiId}`);
+
+    const customDomain = new apigw.DomainName(this, 'Custom Domain', {
+      domainName: (props.stage == "prod") ? `api.${props.project}.${props.companyDomainName}` : `api.${props.stage}.${props.project}.${props.companyDomainName}`,
+      certificate: acm.Certificate.fromCertificateArn(this, "Certificate", `${props.certificateArn}`),
+      endpointType: apigw.EndpointType.EDGE,
+      securityPolicy: apigw.SecurityPolicy.TLS_1_0
+    });
+
+    let apiGatewayCreateBasePathMappingCall: customresource.AwsSdkCall = {
+      service: 'APIGateway',
+      action: 'createBasePathMapping',
+      parameters: {
+        domainName: customDomain.domainName,
+        restApiId: `${props.restApiId}`,
+        basePath: `${props.basePath}`,
+        stage: `${props.stage}`
+      },
+      physicalResourceId: customresource.PhysicalResourceId.of('base-path-mapping-custom-resource')
+    }
+
+    const apiGatewayBasePathMapping = new customresource.AwsCustomResource(this, 'ApiGatewayBasePathMapping', {
```

Comment:
The Base Path Mapping is what I attempted to use at first.  However, one of the properties deploymentStage requires a Stage construct to be passed in and not just the name as a string.  There is no cdk which allows the importing of api stages so I would have to create it in the code.  The issue is the serverless file for pdfconverter already creates the env stage via the usage plan creation, so I can't recreate the stage in cdk since it already exists.  I switched to the sdk call because it allowed me to pass in the deploymentStage as its name as a string like I wanted.